---
title: "Adding Bioconductor Tools in Galaxy"
description: "This is a follow on for the conference, link to training materials from GBCC (see slide)"
authors:
  - name: Maria Doyle
    url: https://github.com/mblue9
  - name: Gretta Yagudayeva
    url: https://github.com/gyagu98
  - name: Charlotte Soneson
    url: https://github.com/csoneson
  - name: Marcel Ramos Pérez
    url: https://github.com/LiNk-NY
date: 2025-06-27
categories: [education, collaborations, galaxy]
image: bioc-galaxy-image.jpeg
toc: true
lightbox: true
---

![](bioc-galaxy-image.jpeg){fig-alt="Bioconductor and Galaxy logos" fig-align="center" width="60%"}

## Introduction

This blog post is a follow-up to the [2025 Galaxy Bioconductor Community Conference](https://gbcc2025.bioconductor.org/about/overview/), held this year at Cold Spring Harbor Laboratory in New York. For the first time, this joint meeting brought together two of the most popular and leading open-source platforms in bioinformatics, Galaxy and Bioconductor, to highlight their powerful synergy and collaborative potential

In this post, we spotlight one of the most exciting sessions from the event: **"Streamlining Bioinformatics: Adding Bioconductor Tools in Galaxy"**, presented by Fabio Cumbo, Jayadev Joshi, Bryan Raubenolt, and Daniel Blankenberg. The session explored how developers can integrate Bioconductor tools into Galaxy by wrapping them within the web-based platform to expand accessibility and usability for researchers.

In case you missed the live session, don't worry-- you can access the full set of training materials, including slides and examples on their [GitHub repository](https://github.com/cumbof/gbcc25_training_session).

![](bioc-galaxy-image3.jpeg){fig-alt="Github linked here" fig-align="center" width="100%"}

## Choosing Galaxy

[Galaxy](https://usegalaxy.org/) is an open source, web-based platform with a graphical user interface (GUI) designed to make complex scientific computational analysis more accessible, reproducible, and transparent.

At its core, Galaxy is a powerful workflow system that empowers users to run bioinformatics tools and pipelines without needing advanced programming or computational training nor high-powered computing clusters.

This becomes particularly valuable for users who lack computing access and/or experience as Galaxy will help you upload and store your data, use tools and pipelines to develop reproducible workflows and answer your most complex scientific inquiry.

![](bioc-galaxy-image2.jpeg){fig-alt="Galaxy Home Page" fig-align="center" width="100%"}

## Using Bioconductor tools in Galaxy

While the conference highlighted the exciting potential of bridging together two of the most powerful open-source platforms in bioinformatics, there are other compelling reasons why this integration goes beyond a technical collaboration.

Bioconductor is a powerful ecosystem of R-based tools that allow for a wide range of analyses, from differential gene expression to advanced pathway enrichment. However, many of these tools remain underutilized, especially outside computational biology, simply because not all researchers are trained in programming nor have the computing access to run such analyses.

**Three key benefits for using Bioconductor tools in Galaxy are:**

1)  [**Benefiting researchers**]{.underline} with increased accessibility for researchers seeking more complex analyses without coding knowledge

2)  [**Benefiting developers**]{.underline} with increased visibility, recognition, and usage for developers as they expand the reach of their work

3)  [**Benefiting Galaxy and Bioconductor**]{.underline} by enabling Galaxy's core values (i.e., accessibility, reproducibility, & transparency) to reach the analysis complexity of Bioconductor

It's not just about ease, it becomes about **equity**, tools like Gene Set Variation Analysis (GSVA), which is essential for functional genomics, should not be limited to researchers with coding skills or powerful computer clusters.

## Wrapping and Integrating Tools into Galaxy

There are several ways to wrap and integrate tools into Galaxy, depending on the tool’s language, dependencies, and how you want users to interact with it. Wrapping a tool means to make it available through Galaxy's GUI for users to run your specific tool without touching the command line. Depending on their *recipe*, see below for a clickable gallery of a portion of the slideshow showcasing the different methods of tool wrapping:

:::::::::::::::: {#awards-gallery .columns}
::: {.column width="18.5%"}
![](bioc-galaxy-4.png){group="slides-gallery" style="border: 1px solid #ccc; padding: 5px;" alt="Slides"}
:::

::: {.column width="18.5%"}
![](bioc-galaxy-5.png){group="slides-gallery" style="border: 1px solid #ccc; padding: 5px;" alt="Slide deck 2"}
:::

::: {.column width="18.5%"}
![](bioc-galaxy-6.png){group="slides-gallery" style="border: 1px solid #ccc; padding: 5px;" alt="Slide deck 3"}
:::

::: {.column width="18.5%"}
![](bioc-galaxy-7.png){group="slides-gallery" style="border: 1px solid #ccc; padding: 5px;" alt="Slide deck 4"}
:::

::: {.column width="18.5%"}
![](bioc-galaxy-8.png){group="slides-gallery" style="border: 1px solid #ccc; padding: 5px;" alt="Slide deck 5"}
:::

::: {.column width="18.5%"}
![](bioc-galaxy-9.png){group="slides-gallery" style="border: 1px solid #ccc; padding: 5px;" alt="Slide deck 6"}
:::

::: {.column width="18.5%"}
![](bioc-galaxy-10.png){group="slides-gallery" style="border: 1px solid #ccc; padding: 5px;" alt="Slide deck 7"}
:::

::: {.column width="18.5%"}
![](bioc-galaxy-11.png){group="slides-gallery" style="border: 1px solid #ccc; padding: 5px;" alt="Slide deck 8"}
:::

::: {.column width="18.5%"}
![](bioc-galaxy-12.png){group="slides-gallery" style="border: 1px solid #ccc; padding: 5px;" alt="Slide deck 9"}
:::

::: {.column width="18.5%"}
![](bioc-galaxy-13.png){group="slides-gallery" style="border: 1px solid #ccc; padding: 5px;" alt="Slide deck 10"}
:::

::: {.column width="18.5%"}
![](bioc-galaxy-14.png){group="slides-gallery" style="border: 1px solid #ccc; padding: 5px;" alt="Slide deck 11"}
:::

::: {.column width="18.5%"}
![](bioc-galaxy-15.png){group="slides-gallery" style="border: 1px solid #ccc; padding: 5px;" alt="Slide deck 12"}
:::

::: {.column width="18.5%"}
![](bioc-galaxy-16.png){group="slides-gallery" style="border: 1px solid #ccc; padding: 5px;" alt="Slide deck 13"}
:::
::::::::::::::::

::: {style="text-align: center;"}
*Slide Deck of all the methods to wrap and integrate Bioconductor tools into Galaxy*
:::

Wrapping and integrating tools into Galaxy can follow multiple paths, from manual configuration to automated wrapper tools.

## Wrapping GSVA in Galaxy as an example

[GSVA](https://www.bioconductor.org/packages/release/bioc/html/GSVA.html#:~:text=Gene%20Set%20Variation%20Analysis%20(GSVA,of%20a%20expression%20data%20set.) is a *a non-parametric, unsupervised method for estimating variation of gene set enrichment through the samples of a expression data set*. 

Adding info about GSVA

GSVA tool as an example, explain it, showing it wrapped put a line of code in to explain it

-R2-G2 (a way to wrap the tools) put a line of code in to explain it
