<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.55">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jeremy Selva">
<meta name="dcterms.date" content="2022-10-22">
<meta name="description" content="This blog post consolidates web links to different Awesome Lists that may be useful to those just starting up in bioinformatics and those experienced in the field.">

<title>Recommended Awesome Lists for Bioconductor Community: Oct 2022 Edition – Bioconductor community blog</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script defer="" data-domain="bioconductor.github.io/biocblog" src="https://plausible.io/js/script.js"></script>


<link rel="stylesheet" href="../../styles.css">
<meta property="og:title" content="Recommended Awesome Lists for Bioconductor Community: Oct 2022 Edition – Bioconductor community blog">
<meta property="og:description" content="This blog post consolidates web links to different Awesome Lists that may be useful to those just starting up in bioinformatics and those experienced in the field.">
<meta property="og:image" content="https://bioconductor.github.io/biocblog/posts/2022-10-22-awesome-lists/awesome.png">
<meta property="og:site_name" content="Bioconductor community blog">
<meta property="og:image:height" content="640">
<meta property="og:image:width" content="1280">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Bioconductor community blog</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../contributing.html"> 
<span class="menu-text">Contributing</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/bioconductor/biocblog"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://genomic.social/@bioconductor"> <i class="bi bi-mastodon" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/company/bioconductor/"> <i class="bi bi-linkedin" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.youtube.com/user/bioconductor"> <i class="bi bi-youtube" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://slack.bioconductor.org/"> <i class="bi bi-slack" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../index.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Recommended Awesome Lists for Bioconductor Community: Oct 2022 Edition</h1>
                  <div>
        <div class="description">
          <p>This blog post consolidates web links to different Awesome Lists that may be useful to those just starting up in bioinformatics and those experienced in the field.</p>
        </div>
      </div>
                          <div class="quarto-categories">
                <div class="quarto-category">education</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p><a href="https://jeremy-selva.netlify.app/">Jeremy Selva</a> </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">October 22, 2022</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">





<section id="introdution" class="level1">
<h1>Introdution</h1>
<p>Initiated by <a href="https://twitter.com/sindresorhus">Sindre Sorhus</a>, an awesome list repository is a list of specific applications or resources that is curated by a domain/field specific community to be useful. They are a great place for those new to the domain to find out what tools others are using. Likewise a curated list of awesome lists can be found in this <a href="https://github.com/topics/awesome">web page</a> in alphabetical order. There is even a <a href="https://twitter.com/awesome__re">Twitter account</a> giving updates to new awesome lists.</p>
<p>However, with more than 5000 public repository of awesome lists, it can be hard to navigate them to find lists that are relevant to the field of bioinformatics and computational biology. This post aims to lighten the work load by providing a manageable list grouped by specific fields of bioinformatics that is of interest to the Bioconductor Community.</p>
</section>
<section id="omics" class="level1">
<h1>Omics</h1>
<p>Omics data is known to be heterogenous, available in numerous forms such as genome, proteome, transcriptome, and metabolome. Recently, imaging or spatial omics have recent joined into the omics family, making multi omics analysis far more challenging. Here we list up some awesome list from the different groups.</p>
<section id="genomics" class="level2">
<h2 class="anchored" data-anchor-id="genomics">Genomics</h2>
<ul>
<li><a href="https://github.com/seandavi/awesome-single-cell" class="uri">https://github.com/seandavi/awesome-single-cell</a></li>
</ul>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-1-contents" aria-controls="callout-1" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Bioconductor Packages
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-1" class="callout-1-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>There are many Bioconductor packages that made it in the <code>awesome-single-cell</code> list.</p>
<ul>
<li><p>RNA-seq</p>
<ul>
<li><p><a href="https://www.bioconductor.org/packages/release/bioc/html/cellTree.html"><code>cellTree</code></a></p></li>
<li><p><a href="https://www.bioconductor.org/packages/release/bioc/html/countsimQC.html"><code>countsimQC</code></a></p></li>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/dittoSeq.html"><code>dittoSeq</code></a></p></li>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/scran.html"><code>scran</code></a></p></li>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/sincell.html"><code>sincell</code></a></p></li>
</ul></li>
<li><p>Quality Control</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/scater.html"><code>scater</code></a></li>
</ul></li>
<li><p>Dimension Reduction</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/destiny.html"><code>destiny</code></a></li>
</ul></li>
<li><p>Cell Projection And Unimodal Integration</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/scmap.html"><code>scmap</code></a></li>
</ul></li>
<li><p>Simulation</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/splatter.html"><code>splatter</code></a></li>
</ul></li>
<li><p>Cell Type Identification and Classification</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/SingleR.html"><code>SingleR</code></a></li>
</ul></li>
<li><p>Copy Number Analysis</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/AneuFinder.html"><code>AneuFinder</code></a></li>
</ul></li>
<li><p>Epigenomics</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/chromVAR.html"><code>chromVAR</code></a></li>
</ul></li>
</ul>
</div>
</div>
</div>
</section>
<section id="proteomics" class="level2">
<h2 class="anchored" data-anchor-id="proteomics">Proteomics</h2>
<ul>
<li><a href="https://github.com/anuragraj/awesome-proteomics" class="uri">https://github.com/anuragraj/awesome-proteomics</a></li>
</ul>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Bioconductor Packages
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Bioconductor package <a href="https://bioconductor.org/packages/3.12/bioc/html/PGA.html"><code>PGA</code></a> is mentioned in the <code>awesome-proteomics</code> list but the package is deprecated on version 3.12 of Bioconductor.</p>
</div>
</div>
</div>
</section>
<section id="lipidomics" class="level2">
<h2 class="anchored" data-anchor-id="lipidomics">Lipidomics</h2>
<ul>
<li><a href="https://github.com/lifs-tools/awesome-lipidomics" class="uri">https://github.com/lifs-tools/awesome-lipidomics</a></li>
</ul>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Bioconductor Packages
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Bioconductor packages found in the <code>awesome-lipidomics</code> lists are</p>
<ul>
<li><p>Programs &amp; Libraries For Lipid Identification</p>
<ul>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/LOBSTAHS.html"><code>LOBSTAHS</code></a></p></li>
<li><p><a href="http://bioconductor.org/packages/release/bioc/html/xcms.html"><code>xcms</code></a></p></li>
</ul></li>
<li><p>Post Processing, Visualization And Statistical Analysis</p>
<ul>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/rgoslin.html"><code>rgoslin</code></a></p></li>
<li><p><a href="https://www.bioconductor.org/packages/release/bioc/html/lipidr.html"><code>lipidr</code></a></p></li>
</ul></li>
<li><p>Databases, Repositories And Other Resources</p>
<ul>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/biodbHmdb.html"><code>biodbHmdb</code></a></p></li>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/biodbLipidmaps.html"><code>biodbLipidMaps</code></a></p></li>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/RMassBank.html"><code>RMassBank</code></a></p></li>
<li><p><a href="https://bioconductor.org/packages/release/data/experiment/html/mtbls2.html"><code>mtbls2</code></a></p></li>
</ul></li>
</ul>
</div>
</div>
</div>
</section>
<section id="spatial-omics" class="level2">
<h2 class="anchored" data-anchor-id="spatial-omics">Spatial Omics</h2>
<ul>
<li><p><a href="https://github.com/crazyhottommy/awesome_spatial_omics" class="uri">https://github.com/crazyhottommy/awesome_spatial_omics</a></p></li>
<li><p><a href="https://github.com/drighelli/awesome-spatial-omics" class="uri">https://github.com/drighelli/awesome-spatial-omics</a></p></li>
</ul>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-4-contents" aria-controls="callout-4" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Bioconductor Packages
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-4" class="callout-4-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Bioconductor packages that made it in <code>crazyhottommy/awesome_spatial_omics</code> lists are</p>
<ul>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/nnSVG.html"><code>nnSVG</code></a></p></li>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/SpotClean.html"><code>SpotClean</code></a></p></li>
</ul>
<p>Bioconductor packages that made it in <code>drighelli/awesome-spatial-omics</code> lists are</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/mistyR.html"><code>mistyR</code></a> It is called <code>MISTy</code> in <code>drighelli</code>’s list.</li>
</ul>
</div>
</div>
</div>
</section>
<section id="multi-omics" class="level2">
<h2 class="anchored" data-anchor-id="multi-omics">Multi-omics</h2>
<ul>
<li><a href="https://github.com/mikelove/awesome-multi-omics" class="uri">https://github.com/mikelove/awesome-multi-omics</a></li>
</ul>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-5-contents" aria-controls="callout-5" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Bioconductor Packages
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-5" class="callout-5-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Bioconductor packages in the <code>awesome-multi-omics</code> lists are</p>
<ul>
<li><p>Multi-omics Correlation or Factor Analysis</p>
<ul>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/pandaR.html"><code>pandaR</code></a></p></li>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/omicade4.html"><code>omicade4</code></a></p></li>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/STATegRa.html"><code>STATegRa</code></a></p></li>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/mogsa.html"><code>mogsa</code></a></p></li>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/mixOmics.html"><code>mixOmics</code></a></p></li>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/fCCAC.html"><code>fCCAC</code></a></p></li>
<li><p><a href="https://www.bioconductor.org/packages/release/bioc/html/MOFA2.html"><code>MOFA2</code></a></p></li>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/combi.html"><code>combi</code></a></p></li>
</ul></li>
<li><p>Multi-Omics Clustering/Classification/Prediction</p>
<ul>
<li><a href="https://www.bioconductor.org/packages/release/bioc/html/iClusterPlus.html"><code>iClusterPlus</code></a></li>
</ul></li>
<li><p>Multi-Omics Networks</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/MOGAMUN.html"><code>MOGAMUN</code></a></li>
</ul></li>
<li><p>Single Cell Multi-Omics</p>
<ul>
<li><p><a href="https://www.bioconductor.org/packages/release/bioc/html/cardelino.html"><code>cardelino</code></a></p></li>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/CiteFuse.html"><code>CiteFuse</code></a></p></li>
</ul></li>
<li><p>Multi-Omics Simulation</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/MOSim.html"><code>MOSim</code></a></li>
</ul></li>
<li><p>Multi-Omics Data Management</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/MultiAssayExperiment.html"><code>MultiAssayExperiment</code></a></li>
</ul></li>
<li><p>Batch Effect Correction</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/MultiBaC.html"><code>MultiBaC</code></a></li>
</ul></li>
</ul>
</div>
</div>
</div>
</section>
</section>
<section id="bioimaging" class="level1">
<h1>Bioimaging</h1>
<p>Bioimaging is useful in helping researcher understand biological processes in real time. Here are a list of useful resources with respect to their specific domain.</p>
<section id="microscopy" class="level2">
<h2 class="anchored" data-anchor-id="microscopy">Microscopy</h2>
<ul>
<li><a href="https://github.com/hallvaaw/awesome-biological-image-analysis" class="uri">https://github.com/hallvaaw/awesome-biological-image-analysis</a></li>
</ul>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-6-contents" aria-controls="callout-6" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Bioconductor Packages
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-6" class="callout-6-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Only Bioconductor package <a href="https://bioconductor.org/packages/release/bioc/html/EBImage.html"><code>EBImage</code></a> is mentioned in the <code>awesome-biological-image-analysis</code> list.</p>
</div>
</div>
</div>
</section>
<section id="medical-images" class="level2">
<h2 class="anchored" data-anchor-id="medical-images">Medical Images</h2>
<ul>
<li><a href="https://github.com/fepegar/awesome-medical-imaging" class="uri">https://github.com/fepegar/awesome-medical-imaging</a></li>
</ul>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
No Bioconductor packages are found in this list
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
</section>
<section id="cytodata" class="level2">
<h2 class="anchored" data-anchor-id="cytodata">Cytodata</h2>
<ul>
<li><a href="https://github.com/cytodata/awesome-cytodata" class="uri">https://github.com/cytodata/awesome-cytodata</a></li>
</ul>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-8-contents" aria-controls="callout-8" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Bioconductor Packages
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-8" class="callout-8-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Only Bioconductor package <a href="https://bioconductor.org/packages/release/bioc/html/EBImage.html"><code>EBImage</code></a> is mentioned in the <code>awesome-cytodata</code> list.</p>
</div>
</div>
</div>
</section>
</section>
<section id="biological-visualisation" class="level1">
<h1>Biological Visualisation</h1>
<p>Visualisation of biological data can be complex. While there are many web based visualisation tools, they can be hard to find on the Internet. Thankfully, <a href="http://markk.co/#/">Mark Keller</a> has consolidated them based on different fields of biological research from genomics, microscopy, to mass spectrometry.</p>
<ul>
<li><p><a href="https://github.com/keller-mark/awesome-biological-visualizations" class="uri">https://github.com/keller-mark/awesome-biological-visualizations</a></p></li>
<li><p><a href="http://genocat.tools/tools/3dviewer/" class="uri">http://genocat.tools/tools/3dviewer/</a></p></li>
</ul>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-9-contents" aria-controls="callout-9" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Bioconductor Packages
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-9" class="callout-9-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Only Bioconductor package <a href="https://bioconductor.org/packages/release/bioc/html/iSEE.html"><code>iSEE</code></a> is mentioned in the <code>awesome-biological-visualizations</code> list.</p>
</div>
</div>
</div>
</section>
<section id="dynamic-reports" class="level1">
<h1>Dynamic Reports</h1>
<p><a href="https://jupyter.org/">Jupyter</a> and <a href="https://quarto.org/">Quarto</a> are few of the most popular tools used in creating dynamic reports that is compatible with several programming languages like <a href="https://www.r-project.org/">R</a> and <a href="https://www.python.org/">Python</a>. Here are some of the links to their awesome lists.</p>
<section id="jupyter" class="level2">
<h2 class="anchored" data-anchor-id="jupyter">Jupyter</h2>
<ul>
<li><p><a href="https://github.com/ml-tooling/best-of-jupyter" class="uri">https://github.com/ml-tooling/best-of-jupyter</a></p></li>
<li><p><a href="https://github.com/markusschanta/awesome-jupyter" class="uri">https://github.com/markusschanta/awesome-jupyter</a></p></li>
</ul>
</section>
<section id="quarto" class="level2">
<h2 class="anchored" data-anchor-id="quarto">Quarto</h2>
<ul>
<li><a href="https://github.com/mcanouil/awesome-quarto" class="uri">https://github.com/mcanouil/awesome-quarto</a></li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The <a href="https://github.com/Bioconductor/biocblog">Bioconductor Community Blog</a> is mentioned.</p>
</div>
</div>
</section>
</section>
<section id="code-version-control" class="level1">
<h1>Code Version Control</h1>
<p>Version control systems (VCS) are software tools that help data analysts manage changes to their programming code over time. One of the most popular VCS tools in use today is <a href="https://git-scm.com/">Git</a>. <a href="https://github.com/">GitHub</a> and <a href="https://about.gitlab.com/">GitLab</a> are one of many cloud-based Git repository hosting services.</p>
<section id="git" class="level2">
<h2 class="anchored" data-anchor-id="git">Git</h2>
<ul>
<li><a href="https://github.com/dictcp/awesome-git" class="uri">https://github.com/dictcp/awesome-git</a></li>
</ul>
</section>
<section id="github" class="level2">
<h2 class="anchored" data-anchor-id="github">GitHub</h2>
<ul>
<li><p><a href="https://github.com/tiimgreen/github-cheat-sheet" class="uri">https://github.com/tiimgreen/github-cheat-sheet</a></p></li>
<li><p><a href="https://github.com/xhacker/awesome-github-extensions" class="uri">https://github.com/xhacker/awesome-github-extensions</a></p></li>
</ul>
</section>
<section id="gitlab" class="level2">
<h2 class="anchored" data-anchor-id="gitlab">GitLab</h2>
<ul>
<li><p><a href="https://gitlab.com/Andorka/awesome-gitlab" class="uri">https://gitlab.com/Andorka/awesome-gitlab</a></p></li>
<li><p><a href="https://gitlab.com/fkromer/awesome-gitlab" class="uri">https://gitlab.com/fkromer/awesome-gitlab</a></p></li>
</ul>
</section>
<section id="gitea" class="level2">
<h2 class="anchored" data-anchor-id="gitea">Gitea</h2>
<ul>
<li><a href="https://gitea.com/gitea/awesome-gitea" class="uri">https://gitea.com/gitea/awesome-gitea</a></li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Unfortunately, I could not find awesome list specific to <a href="https://bitbucket.org/product/">Bitbucket</a> and <a href="https://gitee.com/">Gitee</a></p>
</div>
</div>
</section>
</section>
<section id="reproducible-research" class="level1">
<h1>Reproducible Research</h1>
<p><a href="https://twitter.com/jermdemo">Jeremy Leipzig</a> have created a small <a href="https://github.com/leipzig/awesome-reproducible-research">encyclopedia</a> of reproducible research resources from journals, data repositories, relevant organisations and other awesome lists like the <a href="https://github.com/datasnakes/awesome-reproducible-R">Awesome Reproducible R</a> and <a href="https://github.com/veggiemonk/awesome-docker">Awesome Docker</a> .</p>
<ul>
<li><p><a href="https://github.com/leipzig/awesome-reproducible-research" class="uri">https://github.com/leipzig/awesome-reproducible-research</a></p></li>
<li><p><a href="https://github.com/datasnakes/awesome-reproducible-R" class="uri">https://github.com/datasnakes/awesome-reproducible-R</a></p></li>
<li><p><a href="https://github.com/veggiemonk/awesome-docker" class="uri">https://github.com/veggiemonk/awesome-docker</a></p></li>
</ul>
</section>
<section id="research-software" class="level1">
<h1>Research Software</h1>
<p>It can be daunting task to create a R package as there are many things to do like creating the package skeleton correctly, writing documentation and unit test etc. While there are many R packages created to make this process easily. It is nice to organise these R packages based on their specific roles and <a href="https://twitter.com/patilindrajeets">Indrajeet Patil</a> has done just that.</p>
<p>In addition, it is also important for research software, no matter how useful it is to be deployed in the right registry so that the right group of people can find the software easily. In the case of R, we are blessed with many places to deploy an R package <a href="https://cran.r-project.org/submit.html">CRAN</a>, <a href="https://www.bioconductor.org/">Bioconductor</a>, <a href="https://neuroconductor.org/">Neuroconductor</a>, <a href="https://ropensci.org/r-universe/">R-universe</a>. Similar things are also done on other programming languages as well. The Netherlands eScience Center have created a <a href="https://github.com/NLeSC/awesome-research-software-registries">list</a> of relevant registries based on location, domain and programming language.</p>
<section id="developing-r-packages" class="level2">
<h2 class="anchored" data-anchor-id="developing-r-packages">Developing R packages</h2>
<ul>
<li><a href="https://github.com/IndrajeetPatil/awesome-r-pkgtools" class="uri">https://github.com/IndrajeetPatil/awesome-r-pkgtools</a></li>
</ul>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-12-contents" aria-controls="callout-12" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Bioconductor Packages
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-12" class="callout-12-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Only Bioconductor package <a href="https://bioconductor.org/packages/release/bioc/html/biocthis.html"><code>biocthis</code></a> is mentioned in the <code>awesome-r-pkgtools</code> list.</p>
</div>
</div>
</div>
</section>
<section id="research-software-registry" class="level2">
<h2 class="anchored" data-anchor-id="research-software-registry">Research Software Registry</h2>
<ul>
<li><a href="https://github.com/NLeSC/awesome-research-software-registries" class="uri">https://github.com/NLeSC/awesome-research-software-registries</a></li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The <a href="https://www.bioconductor.org/packages/release/BiocViews.html#___Software">Bioconductor</a> repository is mentioned in the <code>awesome-research-software-registries</code> list.</p>
</div>
</div>
</section>
</section>
<section id="moving-foward" class="level1">
<h1>Moving Foward</h1>
<p>We hope that the information listed above are helpful. The creation of these awesome lists is made possible with the hard work of volunteers and the unconditional help from contributors. If there are any suggestions for improving or new information to add, do let the respective maintainers know.</p>
</section>
<section id="acknowlegments" class="level1">
<h1>Acknowlegments</h1>
<p>This blog post is made possible from <a href="https://www.meetup.com/bioconductor-community-in-asia/events/288949678/">an online Biocondutor meetup</a> for <a href="https://hacktoberfest.com/">Hacktoberfest</a> on 26 October 2022.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/bioconductor\.github\.io\/biocblog\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>