---
title: "Introducing CuratedAtlasQueryR package"
description: |
  CuratedAtlasQueryR is a new package that enables easy programmatic exploration of CELLxGENE single-cell human cell atlas data.
author:
  - name: Maria Doyle
  - name: Stefano Mangiola
date: 2023-02-21
categories:
  - Package
image: "overview.png"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

```{r}
library(CuratedAtlasQueryR)
```
*This blog post was co-written by the Bioconductor Community Manager, Maria Doyle, and the lead developer of CuratedAtlasQueryR, Stefano Mangiola*

We are excited to announce the launch of the [CuratedAtlasQueryR](https://github.com/stemangiola/CuratedAtlasQueryR) package! Millions of cells are now at your fingertips as the harmonized and reannotated CELLxGENE single-cell human atlas can be explored and retrieved as a Bioconductor SingleCellExperiment object. 

![](overview.png)

## Why this package was created

The [Human Cell Atlas](https://www.humancellatlas.org/) is a large-scale single-cell sequencing initiative that aims to map every cell type in the human body. It has the potential to revolutionise our understanding of human cellular biology and the immune system. Data harmonisation, curation and effective data query are essential to extract knowledge from these complex atlases. The [CELLxGENE human cell atlas](https://cellxgene.cziscience.com/) provides an explorable and searchable human atlas. However, [CELLxGENE](https://cellxgene.cziscience.com/) is not harmonised across datasets.

To allow exploration and analyses a cross datasets, tissues and disease through R, we have developed [CuratedAtlasQueryR](https://stemangiola.github.io/CuratedAtlasQueryR/), a cell-resolved query system that allows researchers to select cells based on their type, tissue of origin, and demographics.

## How it differs from existing packages

We use the existing Bioconductor package, [cellxgenedp](https://bioconductor.org/packages/release/bioc/html/cellxgenedp.html) as download source for datasets included in [CELLxGENE](https://cellxgene.cziscience.com/). 

We then harmonised, curated and reannotated the data (immune cell labels). 

- The column classes so they can be represented as unique table.
- We subset the columns that were most common across the atlas (the complete columns for a specific dataset can be retrieved through the [cellxgenedp](https://bioconductor.org/packages/release/bioc/html/cellxgenedp.html) package).
- We consolidated sample identifiers.
- We harmonised tissue labels
- We harmonised cell-type labels under a common ontology
- We provide a consensus-based immune cell label, and a confidence label. Consensus was established among the original annotation and three independent references ([Monaco](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107011), [BLUEPRINT](https://rdrr.io/github/LTLA/celldex/man/BlueprintEncodeData.html), and [Azimuth PBMC](https://azimuth.hubmapconsortium.org/references/human_pbmc/))

We introduced custom column, not present in the original CELLxGENE metadata

- `tissue_harmonised`: a coarser tissue name for better filtering
- `age_days`: the number of days corresponding to the age
- `cell_type_harmonised`: the consensus call identity (for immune cells) using the original and three novel annotations using Seurat Azimuth and SingleR
- `confidence_class`: an ordinal class of how confident `cell_type_harmonised` is. 1 is complete consensus, 2 is 3 out of four and so on.
- `cell_annotation_azimuth_l2`: Azimuth cell annotation
- `cell_annotation_blueprint_singler`: SingleR cell annotation using Blueprint reference
- `cell_annotation_blueprint_monaco`: SingleR cell annotation using Monaco reference
- `sample_id_db`: Sample subdivision for internal use
- `file_id_db`: File subdivision for internal use
- `.sample`: Sample ID
- `.sample_name`: How samples were defined


Differently from [cellxgenedp](https://bioconductor.org/packages/release/bioc/html/cellxgenedp.html), we can query specific cells across datasets based on annotation. the metadata exploration and data download is done on-disk without loading into memory. 

The harmonized and reannotated [CELLxGENE](https://cellxgene.cziscience.com/) single cell human atlas can be explored and retrieved as a Bioconductor [SingleCellExperiment](https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html) object.

We use [DuckDB](https://duckdb.org/) in the package due to it's speed, and low disk imprint in handling large amounts of data.

## How you can use it

**Step 1**: Get started with [CuratedAtlasQueryR](https://github.com/stemangiola/CuratedAtlasQueryR) by exploring the integrated metadata of 28 million cells (on-disk) using [tidyverse](https://www.tidyverse.org/). The atlas includes 344 studies across 40 tissues, providing raw abundances and counts-per-million. ðŸ“Š

```{r, echo=TRUE}
metadata = get_metadata()

metadata
```

**Step 2**: Filter cells of interest using tidyverse, whether it's a specific dataset or a cell type across tissues and diseases. [CuratedAtlasQueryR](https://github.com/stemangiola/CuratedAtlasQueryR) makes it easy to find what you're looking for! ðŸ”

```{r, echo=TRUE}
metadata |>
  dplyr::filter(
      ethnicity == "African" &
      stringr::str_like(assay, "%10x%") &
      tissue == "lung parenchyma" &
      stringr::str_like(cell_type, "%CD4%")
  ) 
```

**Step 3**: Collect the [SingleCellExperiment](https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html) for the cells of interest. Our object uses [HDF5 format](https://en.wikipedia.org/wiki/Hierarchical_Data_Format) to keep the R session light, making it easier to work with larger datasets. ðŸ’»

```{r, echo=TRUE}
metadata |>
  dplyr::filter(
      ethnicity == "African" &
      stringr::str_like(assay, "%10x%") &
      tissue == "lung parenchyma" &
      stringr::str_like(cell_type, "%CD4%")
  ) |>
  get_SingleCellExperiment()
```

**Step 4**: With [CuratedAtlasQueryR](https://github.com/stemangiola/CuratedAtlasQueryR) and [tidySingleCellExperiment](https://stemangiola.github.io/tidySingleCellExperiment/articles/introduction.html), checking the transcription abundance of your favourite gene and cell type across diseases, tissues and hundreds of datasets is just a few lines of code! ðŸŽ‰

```{r, echo=TRUE, eval=FALSE}
library(tidySingleCellExperiment)
library(ggplot2)

metadata |>
  
  # Filter and subset
  filter(cell_type_harmonised=="cd14 mono") |>

  # Get counts per million for NCAM1 gene
  get_SingleCellExperiment(assays = "cpm", features = "HLA-A") |> 
  
  # Plot
  join_features("HLA-A", shape = "wide") |> 
  ggplot(aes( disease, `HLA.A`,color = file_id)) +
  geom_jitter(shape=".") 
```

![](hlaplot1.png)

```{r, echo=TRUE, eval=FALSE}

metadata |> 
    
  # Filter and subset
  filter(cell_type_harmonised=="nk") |> 

  # Get counts per million for NCAM1 gene 
  get_SingleCellExperiment(assays = "cpm", features = "NCAM1") |> 

	# Plot
  join_features("NCAM1", shape = "wide") |> 
  ggplot(aes( tissue_harmonised, NCAM1,color = file_id)) +
  geom_jitter(shape=".") 

```

![](hlaplot2.png)

## Acknowledgements

Thanks and congratulations to Michael Milton [@multimeric](https://github.com/multimeric) (main developer), [Martin Morgan](https://www.roswellpark.org/martin-morgan), [Tony Papenfuss](https://www.wehi.edu.au/people/tony-papenfuss), [Vince Carey](https://vjcitn.github.io/), [Bioconductor](https://www.bioconductor.org/), [Julie Iskander](https://github.com/jIskCoder), [WEHI](https://www.wehi.edu.au/).